### PR TITLE
Dynamic test assert

### PR DESCRIPTION
**Pull request by aeyakovenko**
Fixed bug in TestNode creation that was advertising an unspecified address.  Added an assert to dynamic test that verifies that network converges 10 consecutive rounds.

### CODE REVIEW

File: tests/multinode.rs [hunk 1]
```diff
@@ -410,6 +410,7 @@ fn test_multi_node_dynamic_network() {
         })
         .collect();
 
+    let mut consequtive_success = 0;
```
Comment (garious) [inline tests/multinode.rs]: consecutive

File: src/crdt.rs [hunk 2]
```diff
@@ -151,6 +151,15 @@ impl NodeInfo {
         tpu: SocketAddr,
         tvu_window: SocketAddr,
     ) -> NodeInfo {
+        let daddr: SocketAddr = "0.0.0.0:0".parse().unwrap();
+        for addr in &[ncp, tvu, rpu, tpu, tvu_window] {
+            //make sure that if addr is a dummy address, that its valid
```
Comment (garious) [inline src/crdt.rs]: It's

File: src/crdt.rs [hunk 3]
```diff
@@ -1084,25 +1093,20 @@ pub struct TestNode {
     pub sockets: Sockets,
 }
 
-impl Default for TestNode {
-    fn default() -> Self {
```
Comment (garious) [inline src/crdt.rs]: Rebase fail? Recommended my clippy

File: src/crdt.rs [hunk 4]
```diff
@@ -1084,25 +1093,20 @@ pub struct TestNode {
...
+        let transaction = UdpSocket::bind("127.0.0.1:0").unwrap();
+        let gossip = UdpSocket::bind("127.0.0.1:0").unwrap();
+        let replicate = UdpSocket::bind("127.0.0.1:0").unwrap();
+        let requests = UdpSocket::bind("127.0.0.1:0").unwrap();
+        let repair = UdpSocket::bind("127.0.0.1:0").unwrap();
+
         let gossip_send = UdpSocket::bind("0.0.0.0:0").unwrap();
-        let respond = UdpSocket::bind("0.0.0.0:0").unwrap();
+        let respond = requests.try_clone().unwrap();
         let broadcast = UdpSocket::bind("0.0.0.0:0").unwrap();
```
Comment (garious) [inline src/crdt.rs]: What problem does that solve?

File: src/crdt.rs [hunk 5]
```diff
@@ -151,6 +151,15 @@ impl NodeInfo {
         tpu: SocketAddr,
         tvu_window: SocketAddr,
     ) -> NodeInfo {
+        let daddr: SocketAddr = "0.0.0.0:0".parse().unwrap();
+        for addr in &[ncp, tvu, rpu, tpu, tvu_window] {
+            //make sure that if addr is a dummy address, that its valid
+            if *addr != daddr {
```
Comment (garious) [inline src/crdt.rs]: Could this be moved to a unit test?

**Comment by aeyakovenko** [comment]
@garious this adds a constraint to the types of addresses that can be in `NodeInfo`, seems like an assert is sufficient.  if we return an error, the error needs to be propagated up to all the clients.

**Comment by garious** [comment]
@aeyakovenko, your asserts are on all `pub` fields, no?

**Comment by aeyakovenko** [comment]
@garious yea, you think a better spot for the assert would be in `Crdt::new`?

**Comment by garious** [comment]
"When in Rome" please. All tests in the codebase are in tests. Anyone is generally allowed to make changes that don't break tests. This process content was added in a PR that you approved: https://github.com/solana-labs/solana#code-coverage

**Pull request merged**